### PR TITLE
Allow hiding chats when you are reset

### PR DESCRIPTION
**Pull request by joshblum**
In action:

```
$ kb chat list
[1]  test2                            [] Rekey needed. Waiting for another par... ??? error unboxing chat message: You are not a member of this team (error 2623)                                           
$ kb chat hide test2
Found PRIVATE CHAT conversation: 
$ kb chat list
$ kb chat hide -u test2
Found PRIVATE CHAT conversation: 
$ kb chat list
[1]  test2                            [] Rekey needed. Waiting for another par... ??? error unboxing chat message: You are not a member of this team (error 2623)                                                  
```

### CODE REVIEW

File: go/client/chat_resolver.go [hunk 1]
```diff
@@ -229,7 +226,11 @@ func (r *chatConversationResolver) Resolve(ctx context.Context, req chatConversa
 					info.Visibility, info.Triple.TopicType, info.TopicName, info.TLFNameExpandedSummary())
 			}
 		}
-		return &conversation, false, nil
+		var err error
+		if conversation.Error != nil {
+			err = errors.New(conversation.Error.Message)
+		}
+		return &conversation, false, err
```
Comment (joshblum) [inline go/client/chat_resolver.go]: do we want to log/warn this error here or in the cmd itself?

File: go/client/chat_resolver.go [hunk 2]
```diff
@@ -229,7 +226,11 @@ func (r *chatConversationResolver) Resolve(ctx context.Context, req chatConversa
 					info.Visibility, info.Triple.TopicType, info.TopicName, info.TLFNameExpandedSummary())
 			}
 		}
-		return &conversation, false, nil
+		var err error
+		if conversation.Error != nil {
```
Comment (mmaxim) [inline go/client/chat_resolver.go]: Why bother making this an error at all?

Comment (joshblum) [reply go/client/chat_resolver.go]: I was leaving as is in case other callers depended on it but  i guess we can just remove it

**Comment by mmaxim** [comment]
I'm kind of curious what happens if you try to send into this convo from the CLI after this change, what does it say?

**Comment by joshblum** [comment]
with latest fix back to normal error: 
`▶ ERROR error unboxing chat message: You are not a member of this team (error 2623)`

**Comment by joshblum** [comment]
another bug with chat interactive, now looks like this:

```
$ kb chat read
There are 3 conversations. Please choose one:
[1] private (unverified) test2                                                                                                                                                                              
[2] private test7,test9                                                                                                                                                                                     
[3] private test7                                                                                                                                                                                           
Please enter a number [1-3]: 1
▶ WARNING not exportable error: error unboxing chat message: You are not a member of this team (error 2623) (chat.PermanentUnboxingError)
▶ ERROR GetConversationForCLILocal error: error unboxing chat message: You are not a member of this team (error 2623)
```

**Pull request merged**